Bottom: 2cc8064fcabd936f651294ec2dfb7cb2e58a7f30
Top:    372f7d2f5057fda5752b1a7b6eff3f2f491d7275
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-19 00:30:47 +0100

Add magic flag to stop pour clipping for off-screen buffers

















---

diff --git a/src/create.c b/src/create.c
index df2afc0..5aa4f95 100644
--- a/src/create.c
+++ b/src/create.c
@@ -151,6 +151,7 @@ CreateNewPCB (bool SetDefaultNames)
   ptr = (PCBTypePtr)calloc (1, sizeof (PCBType));
   ptr->Data = CreateNewBuffer ();
   ptr->Data->pcb = (PCBTypePtr) ptr;
+  ptr->Data->ClipPours = 1;
 
   ptr->ThermStyle = 4;
   ptr->IsleArea = 2.e8;
diff --git a/src/global.h b/src/global.h
index 5d163eb..b485a1d 100644
--- a/src/global.h
+++ b/src/global.h
@@ -417,6 +417,7 @@ typedef struct			/* holds all objects */
    *rat_tree;
   struct PCBType *pcb;
   LayerType Layer[MAX_LAYER + 2];	/* add 2 silkscreen layers */
+  int ClipPours;
 } DataType, *DataTypePtr;
 
 typedef struct			/* holds drill information */
diff --git a/src/pour.c b/src/pour.c
index 32b06ac..0e8906a 100644
--- a/src/pour.c
+++ b/src/pour.c
@@ -1389,6 +1389,8 @@ PlowPours (DataType * Data, int type, void *ptr1, void *ptr2,
 void
 RestoreToPours (DataType * Data, int type, void *ptr1, void *ptr2)
 {
+  if (!Data->ClipPours)
+    return;
   if (type == POUR_TYPE)
     {
 #warning FIXME Later: Why do we need to do this?
@@ -1401,6 +1403,8 @@ RestoreToPours (DataType * Data, int type, void *ptr1, void *ptr2)
 void
 ClearFromPours (DataType * Data, int type, void *ptr1, void *ptr2)
 {
+  if (!Data->ClipPours)
+    return;
   if (type == POUR_TYPE)
     {
 #warning FIXME Later: Why do we need to do this?
